### ISSUE TITLE
Test failure on OSX in test_nlmeans_4d_3dsigma_and_threads

### ISSUE THREAD

**Issue by matthew-brett** [nipy/dipy#1265]
```
======================================================================
FAIL: dipy.denoise.tests.test_nlmeans.T
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/buildslave/osx-10.6/dipy-py2_7-osx-10_6/build/venv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/buildslave/osx-10.6/dipy-py2_7-osx-10_6/build/venv/lib/python2.7/site-packages/dipy/denoise/tests/test_nlmeans.py", line 127, in test_nlmeans_4d_3dsigma_and_threads
    assert_equal(duration_all_core < duration_2core, True)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/numpy/testing/utils.py", line 334, in assert_equal
    raise AssertionError(msg)
AssertionError: 
Items are not equal:
 ACTUAL: False
 DESIRED: True
-------------------- >> begin captured stdout << ---------------------
cpu count 4
1
16.0539901257
All
16.1180989742
2
16.1180989742
```

https://nipy.bic.berkeley.edu/builders/dipy-py2.7-osx-10.6/builds/712/steps/shell_9/logs/stdio

Is this a deterministic test?

**Comment by Garyfallidis** [comment]
This test assumes that multi-threading is working on macosx and that omp.h is correctly found.

**Comment by matthew-brett** [comment]
Need to adapt to platforms where omp.h is not found?

**Comment by Garyfallidis** [comment]
Let me check. We are now looking to the other error.

**Comment by arokem** [comment]
Fixed in #1267 

https://nipy.bic.berkeley.edu/builders/dipy-py2.7-osx-10.6/builds/715
